File: csharp/src/ev1/ut3_strings/ejercicios/resueltos/E10_PalindromoSimple_Resuelto.cs
/*
OBJETIVO: Lee una palabra y comprueba si es palindromo ignorando mayusculas y minusculas.
SOLUCION: ver codigo. Se presenta con menu do-while y opcion 0 para salir.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

string opcion;

do
{
    Console.WriteLine("=== E10 Palindromo simple (resuelto) ===");
    Console.WriteLine("1. Comprobar palabra");
    Console.WriteLine("2. Comprobar palabra y mostrar invertida");
    Console.WriteLine("0. Salir");
    Console.Write("Elige una opcion: ");
    opcion = Console.ReadLine() ?? "0";
    Console.WriteLine();

    switch (opcion)
    {
        case "1":
        case "2":
            Console.Write("Escribe una palabra: ");
            string palabra = Console.ReadLine() ?? "";
            string normalizada = palabra.ToLower();
            string invertida = "";

            for (int i = normalizada.Length - 1; i >= 0; i--)
            {
                invertida += normalizada[i];
            }

            bool esPalindromo = normalizada == invertida;

            if (opcion == "2")
            {
                Console.WriteLine("Invertida: " + invertida);
            }

            Console.WriteLine(esPalindromo ? "Es palindromo." : "No es palindromo.");
            break;

        case "0":
            Console.WriteLine("Saliendo del ejercicio.");
            break;

        default:
            Console.WriteLine("Opcion no valida.");
            break;
    }

    Console.WriteLine();
} while (opcion != "0");

File: cpp/src/ev1/ut3_strings/ejercicios/resueltos/E10_PalindromoSimple_Resuelto.cpp
/*
OBJETIVO: Lee una palabra y comprueba si es palindromo ignorando mayusculas y minusculas.
SOLUCION: ver codigo. Se presenta con menu do-while y opcion 0 para salir.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

#include <iostream>
#include <string>
#include <cctype>
using namespace std;

int main() {
    string opcion;

    do {
        cout << "=== E10 Palindromo simple (resuelto) ===" << endl;
        cout << "1. Comprobar palabra" << endl;
        cout << "2. Comprobar palabra y mostrar invertida" << endl;
        cout << "0. Salir" << endl;
        cout << "Elige una opcion: ";
        getline(cin, opcion);
        cout << endl;

        if (opcion == "1" || opcion == "2") {
            cout << "Escribe una palabra: ";
            string palabra;
            getline(cin, palabra);

            string normalizada = palabra;
            for (size_t i = 0; i < normalizada.size(); i++) {
                normalizada[i] = tolower(normalizada[i]);
            }

            bool esPalindromo = true;
            for (size_t i = 0; i < normalizada.size(); i++) {
                if (normalizada[i] != normalizada[normalizada.size() - 1 - i]) {
                    esPalindromo = false;
                    break;
                }
            }

            if (opcion == "2") {
                string invertida = "";
                for (int i = (int)normalizada.size() - 1; i >= 0; i--) {
                    invertida += normalizada[i];
                }
                cout << "Invertida: " << invertida << endl;
            }

            if (esPalindromo) {
                cout << "Es palindromo." << endl;
            } else {
                cout << "No es palindromo." << endl;
            }
        } else if (opcion == "0") {
            cout << "Saliendo del ejercicio." << endl;
        } else {
            cout << "Opcion no valida." << endl;
        }

        cout << endl;
    } while (opcion != "0");

    return 0;
}
